### PR TITLE
mixin for vue apps

### PR DESCRIPTION
**Pull request by rsek**
PR to explore standardizing vue `Application`s of all sorts as a class mixin that wraps FVTT's native `Application`-descended classes.

### CODE REVIEW

File: src/module/actor/sheets/sf-charactersheet.ts [hunk 1]
```diff
@@ -27,19 +26,8 @@ export class StarforgedCharacterSheet extends VueActorSheet {
     return this
   }
 
-  close(...args) {
-    this.actor.moveSheet?.close(...args)
-    return super.close(...args)
-  }
```
Comment (ben) [inline src/module/actor/sheets/sf-charactersheet.ts:33]: So moving this particular functionality to the parent class is great on its own, but now the opening and closing of the move sheet are in different files. How do you feel about the "magic"-ness of that?

File: src/module/actor/sheets/sf-charactersheet.ts [hunk 2]
```diff
@@ -27,19 +26,8 @@ export class StarforgedCharacterSheet extends VueActorSheet {
...
-
   _getHeaderButtons() {
     return [
-      {
-        class: 'ironsworn-toggle-edit-mode',
-        label: 'Edit',
-        icon: 'fas fa-edit',
-        onclick: (e) => this._toggleEditMode(e),
-      },
```
Comment (ben) [inline src/module/actor/sheets/sf-charactersheet.ts:42]: Definitely in favor of moving this up, nice find!

**Review by ben — APPROVED** [review]
I like it. Most of the diff here is moving the edit buttons to the super class, which is a good change. You mentioned in Discord that you want to inline more of the VueRenderHelper code into this new super class, and that sounds good to me. Just one question for you before merging.

File: src/module/actor/sheets/sf-charactersheet.ts (hunk 1 above)
Comment (rsek) [reply src/module/actor/sheets/sf-charactersheet.ts:33]: yeah, i'm not wild about it. i did consider writing a single PC sheet superclass for both to hold both the shared open/close, then making the SF sheet and the IS sheet extensions thereof. but ultimately i decided i wanted to keep this PR simple. it's something i'd like to add very soon, though.

**Pull request merged**